src/api/v1beta1/dynakube_status.go

```diff
@@ -54,9 +57,38 @@ type DynaKubeStatus struct {
 	// Conditions includes status about the current state of the instance
 	Conditions []metav1.Condition `json:"conditions,omitempty"`
 
-	ActiveGate ActiveGateStatus `json:"activeGate,omitempty"`
-	OneAgent   OneAgentStatus   `json:"oneAgent,omitempty"`
-	Synthetic  SyntheticStatus  `json:"synthetic,omitempty"`
+	ActiveGate   ActiveGateStatus   `json:"activeGate,omitempty"`
+	OneAgent     OneAgentStatus     `json:"oneAgent,omitempty"`
+	Synthetic    SyntheticStatus    `json:"synthetic,omitempty"`
+	DynatraceApi DynatraceApiStatus `json:"dynatraceApi,omitempty"`
+}
+
+const MaxRequestInterval = 15 * time.Minute
+
+type DynatraceApiStatus struct {
+	LastTokenProbe                     metav1.Time `json:"lastTokenProbe,omitempty"`
+	LastOneAgentConnectionInfoUpdate   metav1.Time `json:"lastOneAgentConnectionInfoUpdate,omitempty"`
+	LastActiveGateConnectionInfoUpdate metav1.Time `json:"lastActiveGateConnectionInfoUpdate,omitempty"`
+	LastAuthTokenSecretUpdate          metav1.Time `json:"lastAuthTokenSecretUpdate,omitempty"`
+	LastBasicStatusInformationUpdate   metav1.Time `json:"lastBasicStatusInformationUpdate,omitempty"`
+}
+
+func (dynatraceApiStatus *DynatraceApiStatus) ResetCachedTimestamps() {
+	dynatraceApiStatus.LastTokenProbe = metav1.Time{}
+	dynatraceApiStatus.LastOneAgentConnectionInfoUpdate = metav1.Time{}
+	dynatraceApiStatus.LastActiveGateConnectionInfoUpdate = metav1.Time{}
+	dynatraceApiStatus.LastAuthTokenSecretUpdate = metav1.Time{}
+	dynatraceApiStatus.LastBasicStatusInformationUpdate = metav1.Time{}
+}
```

Comment:
move to `..._test.go` file

Reply:
I agree that it would be more convenient to move this function to a test file, as is only used for testing. 
Sadly using functions declared in test files are not available to other packages test files, which is bad as we would need to have code duplication